roofit/roofitcore/src/RooProdPdf.cxx

```diff
@@ -2306,3 +2328,37 @@ Bool_t RooProdPdf::redirectServersHook(const RooAbsCollection& /*newServerList*/
   }
   return kFALSE ;
 }
+
+void RooProdPdf::CacheElem::writeToStream(std::ostream& os) const {
```

Comment:
It seems to me that this function is used mainly for the test program. Should we then keep only there, or do you think is useful in general ? 
If it is useful, then maybe we can add a PrintCache function in RooProdPdf ?

Reply:
I think it is useful in general when debugging a RooProdPdf.

The choice of the name `writeToStream` was made in analogy to [RooArgSet::writeToStream](https://github.com/root-project/root/blob/master/roofit/roofitcore/inc/RooArgSet.h#L104), where the "write" instead of "print" signifies that this is a machine-readable stream where pretty formatting doesn't matter. I didn't call it "PrintCache" also because the usual Print functions in ROOT don't take the outstream as an argument but print directly to cout.